llvm-general/src/LLVM/General/Module.hs

```diff
@@ -5,6 +5,7 @@ module LLVM.General.Module (
     File(..),
 
     withModuleFromAST,
+    astToFFIModule,
```

Comment:
Exposing this function puts the responsibility of understanding how a particular module's memory will be managed into the user - some automatically cleaned up, some not. Linking does pose some challenges, but I'm skeptical this is the right solution.